src/test/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/DiagnosticReportStatusTest.java

```diff
@@ -0,0 +1,81 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+class DiagnosticReportStatusTest {
+
+    private static final String VALID_EDIFACT = "STS++UN";
+    private static final String VALID_EDIFACT_VALUE = "UN";
+
+    @Test
+    void testToEdifactWithValidDiagnosticReportStatus() {
+        final String expected = "STS++UN'";
+
+        final DiagnosticReportStatus diagnosticReportStatus = DiagnosticReportStatus.builder()
+            .event(ReportStatusCode.UNSPECIFIED)
+            .build();
+
+        final String actualValue = diagnosticReportStatus.toEdifact();
+
+        assertEquals(expected, actualValue);
+    }
+
+    @Test
+    void testToEdifactWithDetailsInDiagnosticReportStatus() {
+        final String expected = "STS+Details+UN'";
+
+        final DiagnosticReportStatus diagnosticReportStatus = DiagnosticReportStatus.builder()
+            .detail("Details")
+            .event(ReportStatusCode.UNSPECIFIED)
+            .build();
+
+        final String actualValue = diagnosticReportStatus.toEdifact();
+
+        assertEquals(expected, actualValue);
+    }
+
+    @Test
+    void testBuildWithNullEventThrowsException() {
+        final NullPointerException exception =
+            assertThrows(NullPointerException.class, () -> DiagnosticReportStatus.builder().build());
+
+        assertEquals("event is marked non-null but is null", exception.getMessage());
+    }
+
+    @Test
+    void testFromStringWithValidEdifactStringReturnsDiagnosticReportStatus() {
+        final DiagnosticReportStatus diagnosticReportStatus = DiagnosticReportStatus.fromString(VALID_EDIFACT);
+
+        assertAll(
+            () -> assertEquals("STS", diagnosticReportStatus.getKey()),
+            () -> assertEquals(VALID_EDIFACT_VALUE, diagnosticReportStatus.getValue()),
+            () -> assertEquals("STS++UN'", diagnosticReportStatus.toEdifact())
+
+        );
+    }
+
+    @Test
+    void testFromStringWithDetailsInDiagnosticReportStatusReturnsDiagnosticReportStatus() {
```

Comment:
I feel we could create a few constants as some of the hardcoded values are reused across multiple tests, e.g. `STS+Details+UN`, `STS+Details+UN'`, `STS++UN' `